### PR TITLE
Update release note

### PR DESCRIPTION
**Pull request by VijayanB**
Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>

### Description
Including rc1 release notes and features that are merged after latest odfe release
to 1.0.0.0
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-knn.release-notes-1.0.0.0.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 ## Version 1.0.0.0 Release Notes
 
 Compatible with OpenSearch 1.0.0
+
+### Features
+
+* Refactor interface to support method configuration in field mapping ([#20](https://github.com/opensearch-project/k-NN/pull/20))
+
```
Comment (jmazanec15) [inline release-notes/opensearch-knn.release-notes-1.0.0.0.md:8]: I think you have to manually add additional space types supported. PR was merged in ODFE and then migrated to OpenSearch

Comment (VijayanB) [reply release-notes/opensearch-knn.release-notes-1.0.0.0.md:8]: @jmazanec15 was that part of beta? if so we don't have to add it here.

File: release-notes/opensearch-knn.release-notes-1.0.0.0.md [hunk 2]
```diff
@@ -1,17 +1,38 @@
 ## Version 1.0.0.0 Release Notes
 
 Compatible with OpenSearch 1.0.0
+
+### Features
+
+* Add support for L-inf distance in AKNN, custom scoring and painless scripting ([#315](https://github.com/opendistro-for-elasticsearch/k-NN/pull/315))
```
Comment (jmazanec15) [inline release-notes/opensearch-knn.release-notes-1.0.0.0.md:7]: I believe inter-product space was added as well.

File: release-notes/opensearch-knn.release-notes-1.0.0.0.md (hunk 1 above)
Comment (jmazanec15) [reply release-notes/opensearch-knn.release-notes-1.0.0.0.md:9]: It was, but I think it should be included here too

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

File: release-notes/opensearch-knn.release-notes-1.0.0.0.md (hunk 1 above)
Comment (vamshin) [reply release-notes/opensearch-knn.release-notes-1.0.0.0.md:9]: Any feature after  odfe1.13.2 should be added

File: release-notes/opensearch-knn.release-notes-1.0.0.0.md (hunk 2 above)
Comment (VijayanB) [reply release-notes/opensearch-knn.release-notes-1.0.0.0.md:7]: You are right. It is interesting for some reason it was not picked up by our release drafter. I added it manually by looking over all merged PR.

**Review by jmazanec15 — APPROVED** [review]

**Pull request merged**